src/engraving/layout/layoutsystem.cpp

```diff
@@ -727,6 +727,14 @@ void LayoutSystem::layoutSystemElements(const LayoutOptions& options, LayoutCont
                 Beam* b = cr->beam();
                 b->layout();
             }
+            if (!cr->isChord()) {
+                continue;
+            }
+            for (Chord* grace : toChord(cr)->graceNotes()) {
+                if (grace->beam() && grace->beam()->elements().front() == grace) {
```

Comment:
Please check if the list is empty before calling front()

Reply:
👍

Reply:
Please don't use these endless grace->beam() calls. Add a temp variable and reuse it. This will not only simplify the code but also help to avoid possible performance problems

Reply:
👍👍